manifest/manifest.go

```diff
@@ -34,9 +46,10 @@ func GuessManifestMIMEType(manifest []byte) string {
 	}
 
 	switch meta.MediaType {
-	case DockerV2Schema2MIMEType, DockerV2ListMIMEType: // A recognized type.
+	case DockerV2Schema2MIMEType, DockerV2ListMIMEType, OCIV1DescriptorMIMEType, OCIV1ImageManifestMIMEType, OCIV1ImageManifestListMIMEType, OCIV1ImageSerializationRootfsTarGzipMIMEType, OCIV1ImageSerializationConfigMIMEType: // A recognized type.
 		return meta.MediaType
 	}
+	// TODO(mitr, runcom): flacky? just error out?
```

Comment:
No, this is AFAICS the _only_ way the function can return `DockerV2Schema1MIMEType`, and recognizing that is essential for stripping the JWS signatures = computing the correct manifest digest.